injector/src/main/java/edu/ucr/cs/riple/injector/changes/AddTypeUseMarkerAnnotation.java

```diff
@@ -68,6 +69,13 @@ public Modification computeTextModificationOnType(Type type, AnnotationExpr anno
     if (range == null) {
       return null;
     }
+
+    if (type instanceof ArrayType) {
```

Comment:
@avenger2597 As mentioned previously in this [comment](https://github.com/ucr-riple/NullAwayAnnotator/pull/251#discussion_r1795932716) your changes should be in `findSimpleNameRangeInTypeName` method where we are returning the array's component type as the place to insert / remove annotation for arrays. You should just change the existing case for array and use `type.getTokenRange()` to compute the range right before the first `[`.

Reply:
Done.

Reply:
@avenger2597 Nice, this is much cleaner now.